drop-transfer/src/ws/server/v4.rs

```diff
@@ -539,6 +539,34 @@ impl handler::HandlerLoop for HandlerLoop<'_> {
             .expect("Event channel should always be open");
     }
 
+    async fn finalize_success(self) {
```

Comment:
Could you correct me if my understanding is right:

The `finalize_failure` is called if the handler received any kind of error and `finalize_success` is called if the handler didn't receive any errors. For example, network failure would issue `finalize_failure` and if the file was properly downloaded or canceled then it would be `finalize_success`.

We're relying on the fact that we only have a temporary file left after canceled or unsuccessful task completion?

Reply:
The success is called when the transfer is canceled - meaning peers are not interested in continuing it.
Failure is called in any other cases, mainly per transfer error conditions.

The temps are left in either case, we only delete them when success is called because is case of failure we expect the transfer to be resumed

Reply:
It would be nice to have a central place where we would be explicit about what happens to the files when certain conditions are met.

In this case, we know that the temp file will be deleted in the case of cancel and deleted in a case of success(which will not find the temp file and would fail). I think an improvement could be:
- handler returning some Enum about what to do with the temp file
- handlers themselves deleting the file when they expect it to be done

Reply:
The handler would not fail because we catch the error and handle it properly https://github.com/NordSecurity/libdrop/pull/114/files#diff-fb1fb25ba74414819c4a03c6bc3928d7d57a23ba18feb72833d09b8125058e8cR590

Reply:
The logic is as follows:
* we fetch all possible temp file locations from the DB
* on transfer cancel we try to delete all potential temp files, if the deletion fails (because the file was successfully transferred, or the user deleted it, or whatever), we log the info and continue

Reply:
Re discussion: The failsafe is for successful transfers and not for the files which was the confusion point. In case the transfer was successful but the file handlers didn't clean up properly this is going to clean-up.